auth/gcloud/aio/auth/auth.py

```diff
@@ -191,3 +192,59 @@ def __init__(self, service_file: Optional[str] = None,
         self.access_token_duration = int(content['expires_in'])
         self.access_token_acquired_at = datetime.datetime.now()
         self.acquiring = None
+
+
+SCOPES = ['https://www.googleapis.com/auth/iam']
+
+
+class IamCredentialsClient:
+    def __init__(self, service_file: Optional[str] = None,
+                 session: Optional[aiohttp.ClientSession] = None,
+                 token: Optional[Token] = None) -> None:
+        self.session = session
+        self.token = token or Token(service_file=service_file, session=session, scopes=SCOPES)
+
+        if self.token.token_type != Type.SERVICE_ACCOUNT:
+            raise TypeError('IAM Credentials Client is only valid for use with Service Accounts.')
+
+    async def headers(self):
+        token = await self.token.get()
+        return {
+            'Authorization': f'Bearer {token}',
+        }
+
+    def service_account_email(self):
+        return self.token.service_data.get('client_email')
+
+    async def sign_blob(self, payload: str, service_account_email: Optional[str] = None,
+                        delegates: Optional[list] = None,
+                        session: aiohttp.ClientSession = None,
+                        timeout: int = 10) -> None:
+        if not valid_base64(payload):
+            raise TypeError('payload must be base64 encoded.')
+
+        service_account_email = service_account_email or self.service_account_email()
+
+        if not service_account_email:
+            raise ValueError('sign_blob must have a valid service_account_email')
+
+        resource_name = f'projects/-/serviceAccounts/{service_account_email}'
+        delegates = delegates or [resource_name]
+
+        url = f'https://iamcredentials.googleapis.com/v1/{resource_name}:signBlob'
+
+        json_str = json.dumps({'delegates': delegates, 'payload': payload})
+        headers = await self.headers()
+        headers.update({
+            'Content-Length': str(len(json_str)),
+            'Content-Type': 'application/json'
+        })
+
+        if not self.session:
+            self.session = aiohttp.ClientSession(conn_timeout=10,
+                                                 read_timeout=10)
+        session = session or self.session
+        resp = await session.post(url, data=json_str, headers=headers,
+                                  timeout=timeout)
+        resp.raise_for_status()
+        return await resp.json().get('signedBlob')
```

Comment:
This should be `return (await resp.json()).get('signedBlob')` -- otherwise, this will try to call `.get` on the coroutine and `await` the result.

Reply:
i wonder how this worked so far.... changing.